### PR TITLE
[IMP] website: move facebook page snippet options in left panel 

### PR DESCRIPTION
**Pull request by sdegueldre**
Before this commit, the configuration of the facebook page snippet was
done in a modal window. In an effort to streamline and unify snippet
options, the configuration has been moved into the left panel.

task-2146928

### CODE REVIEW

File: addons/website/static/src/js/editor/snippets.options.js [hunk 1]
```diff
@@ -849,63 +745,117 @@ options.registry.facebookPage = options.Class.extend({
...
     //--------------------------------------------------------------------------
 
     /**
-     * Opens a dialog to configure the facebook page options.
+     * Toggles a tab.
+     *
+     * @see this.selectClass for parameters
+     * @param {String} optionName the name of the option to toggle
+     */
+    toggleOption: function (previewMode, widgetValue, {optionName}) {
```
Comment (qsm-odoo) [inline addons/website/static/src/js/editor/snippets.options.js]: I'd prefer you call the last argument `params` and use the object. Easier to grep if all options use the same argument names.

File: addons/website/static/src/js/editor/snippets.options.js [hunk 2]
```diff
@@ -849,63 +745,117 @@ options.registry.facebookPage = options.Class.extend({
...
      *
      * @see this.selectClass for parameters
      */
-    fbPageOptions: function () {
-        var dialog = new FacebookPageDialog(this, this.fbData).open();
-        dialog.on('save', this, function (fbData) {
-            this.$target.empty();
-            this.fbData = fbData;
-            options.registry.facebookPage.prototype.markFbElement(this, this.$target, this.fbData);
+    pageUrl: function (previewMode, value, params) {
```
Comment (qsm-odoo) [inline addons/website/static/src/js/editor/snippets.options.js]: `widgetValue` for the same reasons

File: addons/website/views/snippets.xml [hunk 3]
```diff
@@ -1191,7 +1185,20 @@
 
     <!-- Facebook Page -->
     <div data-js="facebookPage" data-selector=".o_facebook_page">
-        <we-button data-fb-page-options="true" data-no-preview="true">Facebook Options</we-button>
+        <we-title>Page URL</we-title>
+        <we-input data-page-url="https://www.facebook.com/Odoo" data-no-preview="true"/>
+        <we-title class="o_we_section_title">Tabs</we-title>
+        <we-section class="pl-1 mt-0">
```
Comment (qsm-odoo) [inline addons/website/views/snippets.xml]: Not sure about the style of this, the indentation make the options appear broken to me (but maybe that's just me, but for example I prefer no indentation, no bold title and an underlined title... to check with SBU). In any case:
- this should be implemented like `we-row`. I would name this new feature `we-group`. The title is automatically generated from the string attribute, like `we-row`.
- cannot use pl-1 mt-0 as their spacing depends on the theme, you should add css to the `we-group` element

File: addons/website/views/snippets.xml [hunk 4]
```diff
@@ -1191,7 +1185,20 @@
...
+            <we-checkbox string="Timeline" data-option-name="tab.timeline" data-toggle-option="true" data-no-preview="true"/>
+            <we-checkbox string="Events" data-option-name="tab.events" data-toggle-option="true" data-no-preview="true"/>
+            <we-checkbox string="Messages" data-option-name="tab.messages" data-toggle-option="true" data-no-preview="true"/>
```
Comment (qsm-odoo) [inline addons/website/views/snippets.xml]: You can add the `data-no-preview` on the parent element

File: addons/website/static/src/js/editor/snippets.options.js [hunk 5]
```diff
@@ -849,63 +745,117 @@ options.registry.facebookPage = options.Class.extend({
...
+            } else {
+                this.fbData.tabs = this.fbData.tabs
+                    .split(',')
+                    .filter(t => t !== optionName)
+                    .join(',');
+            }
+        } else {
+            this.fbData[optionName] = widgetValue;
+        }
+        this._markFbElement();
```
Comment (qsm-odoo) [inline addons/website/static/src/js/editor/snippets.options.js]: Mmh... I guess I will have to make option methods async. With this branch is it theoretically possible to break the system.

Comment (sdegueldre) [reply addons/website/static/src/js/editor/snippets.options.js]: should I return the promise of _markFbElement then?

Comment (qsm-odoo) [reply addons/website/static/src/js/editor/snippets.options.js]: Yes as soon as the branch I mentioned in discuss is merged, this will kinda be important

Comment (sdegueldre) [reply addons/website/static/src/js/editor/snippets.options.js]: I made that change preemptively so it should be fine already. Should be ready to merge then.

**Comment by sdegueldre** [comment]
@qsm-odoo This branch has been ported to master, I changed some of the option code around so that the animation doesn't need to be touched. Let me know if I overlooked anything!

File: addons/website/static/src/snippets/s_facebook_page/options.js [hunk 6]
```diff
@@ -141,63 +33,126 @@ options.registry.facebookPage = options.Class.extend({
...
+             this.fbData.height = this.fbData.tabs === 'events' ? 300 : 500;
+            } else if (this.fbData.small_header) {
+             this.fbData.height = this.fbData.show_facepile ? 165 : 70;
+            } else {
+             this.fbData.height = this.fbData.show_facepile ? 225 : 150;
+            }
+            _.each(this.fbData, (value, key) => {
+             this.$target.attr('data-' + key, value);
+             this.$target.data(key, value);
+            });
```
Comment (qsm-odoo) [inline addons/website/static/src/snippets/s_facebook_page/options.js]: Indentation problem, forgot to mention @sdegueldre

File: addons/website/static/src/snippets/s_facebook_page/options.js [hunk 7]
```diff
@@ -141,63 +33,126 @@ options.registry.facebookPage = options.Class.extend({
...
+            this._refreshPublicWidgets(this.$target);
```
[the remark this answers is not in the document]
Comment (sdegueldre) [reply addons/website/static/src/snippets/s_facebook_page/options.js]: Still had to do inside of the willStart returned promise for the widget to load. Any idea why?

**Comment by sbuhl** [comment]
@qsm-odoo and @sdegueldre 
If it's really 2 lines of code you could fix the 2 side effects. But imo, doesn't deserve to lose time on that.

**Comment by sdegueldre** [comment]
@sbuhl @qsm-odoo Fixing the order of the tabs is easy, allowing people to interact with them doesn't seem that easy.

**Pull request closed without merging**